app/bundles/LeadBundle/Entity/LeadListRepository.php

```diff
@@ -365,42 +365,54 @@ public function setDispatcher(EventDispatcherInterface $dispatcher)
     protected function createFilterExpressionSubQuery($table, $alias, $column, $value, array &$parameters, $leadId = null, array $subQueryFilters = [])
     {
         $subQb   = $this->getEntityManager()->getConnection()->createQueryBuilder();
-        $subExpr = $subQb->expr()->andX();
+        $subExpr = null;
 
-        if ('leads' !== $table) {
-            $subExpr->add(
-                $subQb->expr()->eq($alias.'.lead_id', 'l.id')
+        foreach ($subQueryFilters as $subColumn => $subParameter) {
+            if (null === $subExpr) {
+                $subExpr = $subQb->expr()->and($subQb->expr()->eq($subColumn, ":$subParameter"));
+                continue;
+            }
+            $subExpr->with(
+                $subQb->expr()->eq($subColumn, ":$subParameter")
             );
         }
```

Comment:
This seems to be very complex to what we need. I'm wondering whether storing the expressions to an array and then ad them all to the `and()` method would be nicer.

Also we check if the `$expr` is `null` or not in 3 places but on line 407 and 408 we expect that it is an object.

Reply:
@escopecz should be solved by this commit 813aaeecdfad20dea9f4240bd48eff425362fda2